File: python-client-sdk/src/sdk/models/shared/codat_public_api_models_data_datasetpagedresponselinksmodel.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import codat_public_api_models_data_datasetpagedresponsehrefmodel as shared_codat_public_api_models_data_datasetpagedresponsehrefmodel


@dataclass_json
@dataclasses.dataclass
class CodatPublicAPIModelsDataDataSetPagedResponseLinksModel:
    current: Optional[shared_codat_public_api_models_data_datasetpagedresponsehrefmodel.CodatPublicAPIModelsDataDataSetPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_public_api_models_data_datasetpagedresponsehrefmodel.CodatPublicAPIModelsDataDataSetPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_public_api_models_data_datasetpagedresponsehrefmodel.CodatPublicAPIModelsDataDataSetPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_public_api_models_data_datasetpagedresponsehrefmodel.CodatPublicAPIModelsDataDataSetPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsdatadatasetpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatPublicApiModelsDataDataSetPagedResponseHrefModel } from "./codatpublicapimodelsdatadatasetpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatPublicApiModelsDataDataSetPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatPublicApiModelsDataDataSetPagedResponseHrefModel)
  current?: CodatPublicApiModelsDataDataSetPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatPublicApiModelsDataDataSetPagedResponseHrefModel)
  next?: CodatPublicApiModelsDataDataSetPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatPublicApiModelsDataDataSetPagedResponseHrefModel)
  previous?: CodatPublicApiModelsDataDataSetPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatPublicApiModelsDataDataSetPagedResponseHrefModel)
  self?: CodatPublicApiModelsDataDataSetPagedResponseHrefModel;
}